static_file.py

```diff
@@ -138,14 +138,15 @@ class StaticFileTransformationCommand(TransformationCommand):
     urls.
 
     """
-    def __init__(self, static_file, extension='png', commands=None):
+    def __init__(self, static_file, extension=None, commands=None):
         """
         :param static_file: ID of static_file or Active Record
         :param extension: File extension to use
         :param commands: A list of commands (optional)
         """
         self.static_file = static_file
-        self.extension = extension
+        self.extension = extension or \
+            os.path.splitext(static_file.name)[1][1:] or 'png'
```

Comment:
file_path is function field which is dependent on name itself.

https://github.com/fulfilio/nereid/blob/2acbd010c2c00e51cc64aa34630d24da45230c2b/trytond_nereid/static_file.py#L225-L229